asst2/CODE/open_address_ht.c

```diff
@@ -0,0 +1,51 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "open_address_ht.h"
+
+void createTable(void* ptr) {
+  MEM_LIMIT = 8388608/sysconf(_SC_PAGESIZE);
+  VIRT_LIMIT = 25165824/sysconf(_SC_PAGESIZE);
+  HT_SIZE = MEM_LIMIT;
+  ht_entry* table = ptr;
+  for (int i = 0; i < HT_SIZE; ++i) {
+    table[i] = (ht_entry) {HT_NULL_KEY_MASK | HT_NULL_VAL_MASK};
+  }
+}
+
+ht_val ht_put(OpenAddrHashtable table, ht_entry entry) {
```

Comment:
why don't we just add params of pid, where process thinks it is, and where it actually is as args, rather than creating the entry prior? black box it ?